packages/shared/providers.yaml

```diff
@@ -3826,6 +3826,29 @@ ramp-sandbox:
         base_url: https://demo-api.ramp.com
     docs: https://docs.nango.dev/integrations/all/ramp
 
+rapidapi:
+    display_name: RapidAPI
+    categories:
+        - dev-tools
+    auth_mode: API_KEY
+    proxy:
+        headers:
+            x-rapidapi-key: ${apiKey}
+            x-rapidapi-host: ${connectionConfig.host}
+        base_url: https://${connectionConfig.host}
+    docs: https://docs.nango.dev/integrations/all/rapidapi
+    credentials:
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your RapidAPI account
+    connection_config:
+        host:
+            type: string
+            title: Host
+            description: The host of the API (e.g. api-football-v1.p.rapidapi.com)
+            pattern: '^[a-z0-9_-]+$'
```

Comment:
Would something like `'^[a-z0-9_-]+(\.[a-z0-9_-]+)*$'` be better for the pattern to match on the periods?

Reply:
Not sure about this, for ex: here's a sample API key from RapidAPI: `b7c156af2dmgh5c635305f3744bap168553jsp75193c8367ef`

Reply:
But I thought this was the pattern for the host, not the value of the API key?

Reply:
Sorry, completely missed it. Yes you are right, I think this should work for host url.